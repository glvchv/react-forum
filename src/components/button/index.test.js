import React from 'react';
import renderer from 'react-test-renderer';
import Button from '.';
import { BrowserRouter } from 'react-router-dom';

describe('Button component test', () => {
    it('should render the default button', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <Button link="/" type="default" text="test" />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render the delete button', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <Button link="/" type="delete" text="test" />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render the edit button', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <Button link="/" type="edit" text="test" />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

})