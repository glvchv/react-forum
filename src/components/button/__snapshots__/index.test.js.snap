// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test should render the default button 1`] = `
<a
  className="button  default"
  href="/"
  onClick={[Function]}
>
  test
</a>
`;

exports[`Button component test should render the delete button 1`] = `
<a
  className="button  delete"
  href="/"
  onClick={[Function]}
>
  test
</a>
`;

exports[`Button component test should render the edit button 1`] = `
<a
  className="button  edit"
  href="/"
  onClick={[Function]}
>
  test
</a>
`;
